source/taskmanager/src/main/java/nl/aerius/taskmanager/TaskConsumer.java

```diff
@@ -110,12 +114,21 @@ public void messageDeliveryAborted(final Message<MessageMetaData> message, final
     taskMessageHandler.messageDeliveryAborted(message, exception);
   }
 
-  public void start() {
-    executorService.submit(() -> {
+  public synchronized void start() {
+    if (messageHandlerFuture != null && !messageHandlerFuture.isDone()) {
+      try {
+        messageHandlerFuture.get();
+      } catch (final InterruptedException e) {
+        Thread.currentThread().interrupt();
+      } catch (final ExecutionException e) {
+        LOG.trace("TaskConsumer shutdown {} got exception.", taskQueueName, e);
```

Comment:
Might want to use `info` here instead, as it does mean the thread stopped with a runtime exception or something like it which can be valuable info in production (even if it's not error worthy)?